code/lab/dialogs/lab_ui.cpp

```diff
@@ -1233,13 +1233,104 @@ void LabUi::show_object_options() const
 				if (getLabManager()->isSafeForShips()) {
 					if (Button("Destroy ship")) {
 						if (Objects[getLabManager()->CurrentObject].type == OBJ_SHIP) {
+							// If we have an undocker, delete it before destroying the current ship
+							getLabManager()->deleteDockerObject();
+
 							auto obj = &Objects[getLabManager()->CurrentObject];
 
 							obj->flags.remove(Object::Object_Flags::Player_ship);
 							ship_self_destruct(obj);
 						}
 					}
 
+					const ship* dockee_shipp = &Ships[Objects[getLabManager()->CurrentObject].instance];
+					auto dockee_dock_map = get_docking_point_map(Ship_info[dockee_shipp->ship_info_index].model_num);
+
+					if (!dockee_dock_map.empty()) {
+
+						if (ImGui::BeginCombo("Docker Ship Class", Ship_info[getLabManager()->DockerClass].name)) {
+							for (size_t i = 0; i < Ship_info.size(); ++i) {
+								bool is_selected = (static_cast<int>(i) == getLabManager()->DockerClass);
+								if (ImGui::Selectable(Ship_info[i].name, is_selected)) {
+									getLabManager()->DockerClass = static_cast<int>(i);
+									// Load model if needed
+									auto& dsip = Ship_info[getLabManager()->DockerClass];
+									if (dsip.model_num < 0) {
+										dsip.model_num = model_load(dsip.pof_file, &dsip);
+									}
+									auto new_dock_map = get_docking_point_map(dsip.model_num);
+
+									// Auto-select first available dockpoint (or clear if none)
+									if (!new_dock_map.empty()) {
+										getLabManager()->DockerDockPoint = new_dock_map.begin()->second;
+									} else {
+										getLabManager()->DockerDockPoint.clear();
+									}
+								}
+								if (is_selected)
+									ImGui::SetItemDefaultFocus();
+							}
+							ImGui::EndCombo();
+						}
+
+						auto& dsip = Ship_info[getLabManager()->DockerClass];
+						if (dsip.model_num < 0) {
+							dsip.model_num = model_load(dsip.pof_file, &dsip);
+						}
+						auto dock_map = get_docking_point_map(dsip.model_num);
+
+						// Ensure DockerDockPoint is initialized once based on the current DockerClass
+						if (getLabManager()->DockerDockPoint.empty()) {
+							if (!dock_map.empty()) {
+								getLabManager()->DockerDockPoint = dock_map.begin()->second;
+							}
+						}
+
+						const char* docker_label = getLabManager()->DockerDockPoint.c_str();
+
+						if (ImGui::BeginCombo("Docker Dockpoint", docker_label)) {
+							if (!dock_map.empty()) {
+								for (const auto& [index, name] : dock_map) {
+									bool is_selected = (name == getLabManager()->DockerDockPoint);
+									if (ImGui::Selectable(name.c_str(), is_selected)) {
+										getLabManager()->DockerDockPoint = name;
+									}
+									if (is_selected)
+										ImGui::SetItemDefaultFocus();
```

Comment:
I don't *really* care that much, but you mixed braces around the if block on the previous check but not here and not in some later checks.  Looks like it happened below, as well.